### PR TITLE
SOLR-12955: separate DistribUpdateProcessor to Zk and nonZk classes

### PR DESCRIPTION
**Pull request by barrotsteindev**

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/update/processor/CdcrUpdateProcessor.java [hunk 1]
```diff
@@ -16,115 +16,12 @@
...
-import org.apache.solr.update.UpdateCommand;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 /**
  * <p>
- * Extends {@link org.apache.solr.update.processor.DistributedUpdateProcessor} to force peer sync logic
- * for every updates. This ensures that the version parameter sent by the source cluster is kept
- * by the target cluster.
+ * An Interface for Shared params for Cdcr Update Processors
```
[the remark this answers is not in the document]
Comment (barrotsteindev) [reply solr/core/src/java/org/apache/solr/update/processor/CdcrUpdateProcessor.java]: I will revert this.

**Comment by barrotsteindev** [comment]
I will rebase this PR on the latest master.
It is going to be quite hard to separate this into smaller changes, since I tried not to change the core logic where possible, and instead remove the if clauses(if zkEnabled) and disjoin them into two distinct classes.Alas,

**Comment by barrotsteindev** [comment]
@dsmiley, would you want this to be separated to different commits on the development branch, so you can view each one and review them separately, or would you like to commit each one to master and ensure no tests failure occurs after each commit?
I have started hacking away on a separate branch, and the PR can be viewed [here](https://github.com/apache/lucene-solr/pull/519).

**Comment by dsmiley** [comment]
There is no "development" branch, so I don't know what you mean by that.  You don't _have_ to run the full test suite after each commit, but _ideally_ it ought to pass on each commit if it were to be run.  I don't think I will do it.  It's up to you to pause on each commit for my input but I don't think necessary since I'm overall happy with the final result, I just want to see the incremental steps to observe where the code is going at each juncture.  When I finally commit something, I'm not sure yet if I'm going to do it in one commit or do it separately... but don't let that stop you from breaking it into as small steps as you please.  Please use commit comments to point out anything non-obvious.  The purpose here is mostly to show what went where since a single all-in-one change is kinda unreviewable (or at least would require committing in blind faith).

**Comment by barrotsteindev** [comment]
Opened #519

**Pull request closed without merging**